Synthesis failed for nodejs-bigquery

Hello! Autosynth couldn't regenerate nodejs-bigquery. :broken_heart:

Here's the output from running `synth.py`:

```
Cloning into 'working_repo'...
Switched to a new branch 'autosynth'
Cloning into '/tmpfs/tmp/tmpwcb58jxo/synthtool'...
Switched to branch 'autosynth-self'
Note: checking out '1ca9f07c534cf500b7ae7b21b87f23607e0aa9de'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 1ca9f07 chore: add Blunderbuss config (#688)
Note: checking out '19465d3ec5e5acdb01521d8f3bddd311bcbee28d'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 19465d3 build: use codecov's action, now that it's authless (#499)
Switched to a new branch 'autosynth-self-1'
2020-05-06 04:37:07 [INFO] Running synthtool
2020-05-06 04:37:07 [INFO] ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-05-06 04:37:07,695 synthtool > Executing /tmpfs/src/github/synthtool/working_repo/synth.py.
On branch autosynth-self-1
nothing to commit, working tree clean
2020-05-06 04:37:07,823 synthtool > Using precloned repo /tmpfs/tmp/tmpwcb58jxo/synthtool
.eslintignore
.eslintrc.json
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/publish.yml
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/lint.cfg
.kokoro/continuous/node10/samples-test.cfg
.kokoro/continuous/node10/system-test.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/continuous/node8/common.cfg
.kokoro/continuous/node8/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node10/docs.cfg
.kokoro/presubmit/node10/lint.cfg
.kokoro/presubmit/node10/samples-test.cfg
.kokoro/presubmit/node10/system-test.cfg
.kokoro/presubmit/node10/test.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/presubmit/node8/common.cfg
.kokoro/presubmit/node8/test.cfg
.kokoro/presubmit/windows/common.cfg
.kokoro/presubmit/windows/test.cfg
.kokoro/publish.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
renovate.json
samples/README.md
2020-05-06 04:37:07,961 synthtool > Installing dependencies...
npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/bigquery@4.7.0 prepare /tmpfs/src/github/synthtool/working_repo
> npm run compile

npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/bigquery@4.7.0 precompile /tmpfs/src/github/synthtool/working_repo
> gts clean

Removing build ...

> @google-cloud/bigquery@4.7.0 compile /tmpfs/src/github/synthtool/working_repo
> tsc -p . && cp src/types.d.ts build/src/

[96mtest/bigquery.ts[0m:[93m182[0m:[93m29[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'Service'.

[7m182[0m       const calledWith = bq.calledWith_[0];
[7m   [0m [91m                            ~~~~~~~~~~~[0m

[96mtest/dataset.ts[0m:[93m121[0m:[93m29[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m121[0m       const calledWith = ds.calledWith_[0];
[7m   [0m [91m                            ~~~~~~~~~~~[0m

[96mtest/job.ts[0m:[93m114[0m:[93m37[0m - [91merror[0m[90m TS2339: [0mProperty 'parent' does not exist on type '{}'.

[7m114[0m       assert.strictEqual(calledWith.parent, BIGQUERY);
[7m   [0m [91m                                    ~~~~~~[0m

[96mtest/job.ts[0m:[93m115[0m:[93m37[0m - [91merror[0m[90m TS2339: [0mProperty 'baseUrl' does not exist on type '{}'.

[7m115[0m       assert.strictEqual(calledWith.baseUrl, '/jobs');
[7m   [0m [91m                                    ~~~~~~~[0m

[96mtest/job.ts[0m:[93m116[0m:[93m37[0m - [91merror[0m[90m TS2339: [0mProperty 'id' does not exist on type '{}'.

[7m116[0m       assert.strictEqual(calledWith.id, JOB_ID);
[7m   [0m [91m                                    ~~[0m

[96mtest/job.ts[0m:[93m117[0m:[93m41[0m - [91merror[0m[90m TS2339: [0mProperty 'methods' does not exist on type '{}'.

[7m117[0m       assert.deepStrictEqual(calledWith.methods, {
[7m   [0m [91m                                        ~~~~~~~[0m

[96mtest/table.ts[0m:[93m220[0m:[93m32[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m220[0m       const calledWith = table.calledWith_[0];
[7m   [0m [91m                               ~~~~~~~~~~~[0m

[96mtest/table.ts[0m:[93m2336[0m:[93m39[0m - [91merror[0m[90m TS2339: [0mProperty 'raw' does not exist on type '{ rows: { insertId: number; json: { state: string; gender: string; year: string; name: string; count: string; }; }[]; }'.

[7m2336[0m       assert.strictEqual(reqOpts.json.raw, undefined);
[7m    [0m [91m                                      ~~~[0m


Found 8 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/bigquery@4.7.0 compile: `tsc -p . && cp src/types.d.ts build/src/`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/bigquery@4.7.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-06T11_37_27_997Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/bigquery@4.7.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/bigquery@4.7.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-06T11_37_28_035Z-debug.log
2020-05-06 04:37:28,064 synthtool > Failed executing npm install:

None
2020-05-06 04:37:28,065 synthtool > Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/tmpfs/src/github/synthtool/working_repo/synth.py", line 14, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 166, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-05-06 04:37:28 [ERROR] Synthesis failed
HEAD is now at 1ca9f07 chore: add Blunderbuss config (#688)
Switched to branch 'autosynth-self'
Note: checking out '1ca9f07c534cf500b7ae7b21b87f23607e0aa9de'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at 1ca9f07 chore: add Blunderbuss config (#688)
Previous HEAD position was 19465d3 build: use codecov's action, now that it's authless (#499)
HEAD is now at ab88356 fix: make .kokoro-autosynth executable (#522)
Switched to a new branch 'autosynth-17'
2020-05-06 04:37:28 [INFO] Running synthtool
2020-05-06 04:37:28 [INFO] ['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']
2020-05-06 04:37:28,363 synthtool > Executing /tmpfs/src/github/synthtool/working_repo/synth.py.
On branch autosynth-17
nothing to commit, working tree clean
2020-05-06 04:37:28,499 synthtool > Using precloned repo /tmpfs/tmp/tmpwcb58jxo/synthtool
.eslintignore
.eslintrc.json
.github/ISSUE_TEMPLATE/bug_report.md
.github/ISSUE_TEMPLATE/feature_request.md
.github/ISSUE_TEMPLATE/support_request.md
.github/PULL_REQUEST_TEMPLATE.md
.github/publish.yml
.github/release-please.yml
.github/workflows/ci.yaml
.kokoro/common.cfg
.kokoro/continuous/node10/common.cfg
.kokoro/continuous/node10/docs.cfg
.kokoro/continuous/node10/lint.cfg
.kokoro/continuous/node10/samples-test.cfg
.kokoro/continuous/node10/system-test.cfg
.kokoro/continuous/node10/test.cfg
.kokoro/continuous/node12/common.cfg
.kokoro/continuous/node12/test.cfg
.kokoro/continuous/node8/common.cfg
.kokoro/continuous/node8/test.cfg
.kokoro/docs.sh
.kokoro/lint.sh
.kokoro/presubmit/node10/common.cfg
.kokoro/presubmit/node10/docs.cfg
.kokoro/presubmit/node10/lint.cfg
.kokoro/presubmit/node10/samples-test.cfg
.kokoro/presubmit/node10/system-test.cfg
.kokoro/presubmit/node10/test.cfg
.kokoro/presubmit/node12/common.cfg
.kokoro/presubmit/node12/test.cfg
.kokoro/presubmit/node8/common.cfg
.kokoro/presubmit/node8/test.cfg
.kokoro/presubmit/windows/common.cfg
.kokoro/presubmit/windows/test.cfg
.kokoro/publish.sh
.kokoro/release/docs.cfg
.kokoro/release/docs.sh
.kokoro/release/publish.cfg
.kokoro/samples-test.sh
.kokoro/system-test.sh
.kokoro/test.bat
.kokoro/test.sh
.kokoro/trampoline.sh
.mocharc.js
.nycrc
.prettierignore
.prettierrc.js
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
README.md
renovate.json
samples/README.md
2020-05-06 04:37:28,638 synthtool > Installing dependencies...
npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/bigquery@4.7.0 prepare /tmpfs/src/github/synthtool/working_repo
> npm run compile

npm WARN npm npm does not support Node.js v12.16.3
npm WARN npm You should probably upgrade to a newer version of node as we
npm WARN npm can't make any promises that npm will work with this version.
npm WARN npm Supported releases of Node.js are the latest release of 6, 8, 9, 10, 11.
npm WARN npm You can find the latest version at https://nodejs.org/

> @google-cloud/bigquery@4.7.0 precompile /tmpfs/src/github/synthtool/working_repo
> gts clean

Removing build ...

> @google-cloud/bigquery@4.7.0 compile /tmpfs/src/github/synthtool/working_repo
> tsc -p . && cp src/types.d.ts build/src/

[96mtest/bigquery.ts[0m:[93m182[0m:[93m29[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'Service'.

[7m182[0m       const calledWith = bq.calledWith_[0];
[7m   [0m [91m                            ~~~~~~~~~~~[0m

[96mtest/dataset.ts[0m:[93m121[0m:[93m29[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m121[0m       const calledWith = ds.calledWith_[0];
[7m   [0m [91m                            ~~~~~~~~~~~[0m

[96mtest/job.ts[0m:[93m114[0m:[93m37[0m - [91merror[0m[90m TS2339: [0mProperty 'parent' does not exist on type '{}'.

[7m114[0m       assert.strictEqual(calledWith.parent, BIGQUERY);
[7m   [0m [91m                                    ~~~~~~[0m

[96mtest/job.ts[0m:[93m115[0m:[93m37[0m - [91merror[0m[90m TS2339: [0mProperty 'baseUrl' does not exist on type '{}'.

[7m115[0m       assert.strictEqual(calledWith.baseUrl, '/jobs');
[7m   [0m [91m                                    ~~~~~~~[0m

[96mtest/job.ts[0m:[93m116[0m:[93m37[0m - [91merror[0m[90m TS2339: [0mProperty 'id' does not exist on type '{}'.

[7m116[0m       assert.strictEqual(calledWith.id, JOB_ID);
[7m   [0m [91m                                    ~~[0m

[96mtest/job.ts[0m:[93m117[0m:[93m41[0m - [91merror[0m[90m TS2339: [0mProperty 'methods' does not exist on type '{}'.

[7m117[0m       assert.deepStrictEqual(calledWith.methods, {
[7m   [0m [91m                                        ~~~~~~~[0m

[96mtest/table.ts[0m:[93m220[0m:[93m32[0m - [91merror[0m[90m TS2339: [0mProperty 'calledWith_' does not exist on type 'ServiceObject<any>'.

[7m220[0m       const calledWith = table.calledWith_[0];
[7m   [0m [91m                               ~~~~~~~~~~~[0m

[96mtest/table.ts[0m:[93m2336[0m:[93m39[0m - [91merror[0m[90m TS2339: [0mProperty 'raw' does not exist on type '{ rows: { insertId: number; json: { state: string; gender: string; year: string; name: string; count: string; }; }[]; }'.

[7m2336[0m       assert.strictEqual(reqOpts.json.raw, undefined);
[7m    [0m [91m                                      ~~~[0m


Found 8 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/bigquery@4.7.0 compile: `tsc -p . && cp src/types.d.ts build/src/`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/bigquery@4.7.0 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-06T11_37_36_971Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/bigquery@4.7.0 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/bigquery@4.7.0 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2020-05-06T11_37_36_987Z-debug.log
2020-05-06 04:37:37,002 synthtool > Failed executing npm install:

None
2020-05-06 04:37:37,003 synthtool > Wrote metadata to synth.metadata.
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/tmpfs/src/github/synthtool/working_repo/synth.py", line 14, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 166, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2020-05-06 04:37:37 [ERROR] Synthesis failed
HEAD is now at 1ca9f07 chore: add Blunderbuss config (#688)
Switched to branch 'autosynth'
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 584, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 465, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 574, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 360, in synthesize_loop
    synthesize_inner_loop(toolbox, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 370, in synthesize_inner_loop
    synthesizer, len(toolbox.versions) - 1
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 115, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](https://sponge/d5e01c0b-0d2d-4da4-b495-86674ea6021f).

--- a/test/bigquery.ts
+++ b/test/bigquery.ts
@@ -48,6 +48,14 @@ class FakeApiError {
   }
 }
 
+interface CalledWithService extends Service {
+  calledWith_: Array<{
+    baseUrl: string;
+    scopes: string[];
+    packageJson: {};
+  }>;
+}
+
 let promisified = false;
 const fakePfy = extend({}, pfy, {
   promisifyAll: (c: Function, options: pfy.PromisifyAllOptions) => {
@@ -179,7 +187,7 @@ describe('BigQuery', () => {
     it('should inherit from Service', () => {
       assert(bq instanceof Service);
 
-      const calledWith = bq.calledWith_[0];
+      const calledWith = (bq as CalledWithService).calledWith_[0];
 
       const baseUrl = 'https://bigquery.googleapis.com/bigquery/v2';
       assert.strictEqual(calledWith.baseUrl, baseUrl);

--- a/test/dataset.ts
+++ b/test/dataset.ts
@@ -29,6 +29,15 @@ import * as _root from '../src';
 import {DatasetOptions} from '../src/dataset';
 import {FormattedMetadata, TableOptions} from '../src/table';
 
+interface CalledWithDataset extends ServiceObject {
+  calledWith_: Array<{
+    parent: {};
+    baseUrl: string;
+    id: string;
+    methods: string[];
+  }>;
+}
+
 let promisified = false;
 const fakePfy = extend({}, pfy, {
   promisifyAll: (c: Function, options: pfy.PromisifyAllOptions) => {
@@ -118,7 +127,7 @@ describe('BigQuery/Dataset', () => {
     it('should inherit from ServiceObject', () => {
       assert(ds instanceof ServiceObject);
 
-      const calledWith = ds.calledWith_[0];
+      const calledWith = (ds as CalledWithDataset).calledWith_[0];
 
       assert.strictEqual(calledWith.parent, BIGQUERY);
       assert.strictEqual(calledWith.baseUrl, '/datasets');

--- a/test/job.ts
+++ b/test/job.ts
@@ -36,6 +36,15 @@ class FakeOperation {
   }
 }
 
+interface CalledWithJob extends FakeOperation {
+  calledWith_: Array<{
+    parent: {};
+    baseUrl: string;
+    id: string;
+    methods: string[];
+  }>;
+}
+
 let promisified = false;
 const fakePfy = extend({}, pfy, {
   promisifyAll: (c: Function) => {
@@ -109,7 +118,7 @@ describe('BigQuery/Job', () => {
     it('should inherit from Operation', () => {
       assert(job instanceof FakeOperation);
 
-      const calledWith = job.calledWith_[0];
+      const calledWith = (job as CalledWithJob).calledWith_[0];
 
       assert.strictEqual(calledWith.parent, BIGQUERY);
       assert.strictEqual(calledWith.baseUrl, '/jobs');

--- a/test/table.ts
+++ b/test/table.ts
@@ -45,6 +45,15 @@ import {
 } from '../src/table';
 import bigquery from '../src/types';
 
+interface CalledWithTable extends ServiceObject {
+  calledWith_: Array<{
+    parent: {};
+    baseUrl: string;
+    id: string;
+    methods: string[];
+  }>;
+}
+
 let promisified = false;
 let makeWritableStreamOverride: Function | null;
 let isCustomTypeOverride: Function | null;
@@ -217,7 +226,7 @@ describe('BigQuery/Table', () => {
       const table = new Table(datasetInstance, TABLE_ID);
       assert(table instanceof ServiceObject);
 
-      const calledWith = table.calledWith_[0];
+      const calledWith = (table as CalledWithTable).calledWith_[0];
 
       assert.strictEqual(calledWith.parent, datasetInstance);
       assert.strictEqual(calledWith.baseUrl, '/tables');
@@ -2333,7 +2342,6 @@ describe('BigQuery/Table', () => {
       assert.strictEqual(reqOpts.method, 'POST');
       assert.strictEqual(reqOpts.uri, '/insertAll');
       assert.deepStrictEqual(reqOpts.json, {rows: rawData});
-      assert.strictEqual(reqOpts.json.raw, undefined);
     });
 
     it('should accept options', async () => {

